### PR TITLE
fix: detect minimum swap value by the trader plugin

### LINKED ISSUE #2679: [Bug] Detect minimum swap value by the trader plugin
## Catalogue

- [ ] UI/UX
- [ ] Mask Core Feature
- [x] Mask Plugin Feature
- [ ] Security/Cryptography Feature
- [ ] Other
  - /_List it here_/

## Part 1: Introduction and Background

+ Refactor [validateMessage](https://github.com/DimensionDev/Maskbook/blob/418df6934e810d55bde96e1384310e71f8a04e28/packages/maskbook/src/plugins/Trader/UI/trader/TradeForm.tsx#L218) to be able to validate trade amounts.
+ Let's define the minimum amount value to `0.000001`. If an amount is less than that value should return `Input/Output amount is below the minimum amount.` as error message.

## Part 3: Expected Behavior

Here is the pic from myetherwallet.

image

### PR DESCRIPTION
**Pull request by lelenei**
closes #2679

### CODE REVIEW

**Review by Jack-Works — APPROVED** [review]

File: packages/maskbook/src/_locales/en/messages.json [hunk 1]
```diff
@@ -409,6 +409,7 @@
     "plugin_trader_settings_data_source_secondary": "Select the source of cryptocurrencies' trending data.",
     "plugin_trader_settings_trade_provider_primary": "Choose Exchange Network",
     "plugin_trader_settings_trade_provider_secondary": "Swap cryptocurrencies on different exchange networks.",
+    "plugin_trade_error_amount_less_minimum_amount": "Input/Output amount is below the minimum amount.",
```
Comment (guanbinrui) [inline packages/maskbook/src/_locales/en/messages.json:412]: They are two different situiation:

+ Input amount is below the minimum amount.
+ Output amount is below the minimum amount.

File: packages/maskbook/src/plugins/Trader/constants/trending.ts [hunk 2]
```diff
@@ -16,6 +16,8 @@ export const THIRD_PARTY_V1_BASE_URL = 'https://3rdparty-apis.coinmarketcap.com/
 // the bitcoin ledger started at 03 Jan 2009
 export const BTC_FIRST_LEGER_DATE = new Date('2009-01-03T00:00:00.000Z')
 
+export const MINIMUM_AMOUNT = 0.000001
```
Comment (guanbinrui) [inline packages/maskbook/src/plugins/Trader/constants/trending.ts:19]: Move this to `packages/maskbook/src/plugins/Trader/constants/trader.ts`.

File: packages/maskbook/src/plugins/Trader/UI/trader/TradeForm.tsx [hunk 3]
```diff
@@ -199,6 +200,10 @@ export function TradeForm(props: TradeFormProps) {
     const validationMessage = useMemo(() => {
         if (inputTokenTradeAmount.isZero() && outputTokenTradeAmount.isZero())
             return t('plugin_trader_error_amount_absence')
+        if (inputTokenTradeAmount.isLessThan(MINIMUM_AMOUNT))
```
Comment (guanbinrui) [inline packages/maskbook/src/plugins/Trader/UI/trader/TradeForm.tsx:203]: WRONG! Please recheck how is the trade amount made. You cannot compare it with a fraction number.

File: packages/maskbook/src/plugins/Trader/constants/trader.ts [hunk 4]
```diff
@@ -3,6 +3,8 @@ import { CONSTANTS } from '../../../web3/constants'
 import { createERC20Token, getConstant } from '../../../web3/helpers'
 import { ChainId, ERC20TokenDetailed } from '../../../web3/types'
 
+export const MINIMUM_AMOUNT = 0.000001
```
Comment (septs) [inline packages/maskbook/src/plugins/Trader/constants/trader.ts:6]:
```suggestion
export const MINIMUM_AMOUNT = new BigNumber('1').shiftedBy(-6)
```

```
> const BigNumber = require('bignumber.js')
> new BigNumber('1').shiftedBy(-6).toString()
'0.000001'
```

File: packages/maskbook/src/plugins/Trader/UI/trader/TradeForm.tsx (hunk 3 above)
Comment (lelenei) [reply packages/maskbook/src/plugins/Trader/UI/trader/TradeForm.tsx:203]: BigNumber can handle data accuracy

**Comment by guanbinrui** [comment]
Hi, @lelenei Don't forget to hit the `Ready for review` button.

**Review by guanbinrui — APPROVED** [review]

**Review by Jack-Works — APPROVED** [review]

**Pull request merged**